dvc/command/metrics.py

```diff
@@ -12,7 +12,46 @@ def _show(self, metrics):
                 logger.info("{}:".format(branch))
 
             for fname, metric in val.items():
-                logger.info("\t{}: {}".format(fname, metric))
+                self._output_formatted_metric(fname, metric)
+
+    def _is_json(self, content):
+        import json
+
+        try:
+            json.loads(content)
+            return True
+        except ValueError:
+            return False
+
+    def _output_formatted_metric(self, fname, metric):
+        import csv
+
+        if "\n" not in metric:
+            logger.info("\t{}: {}".format(fname, metric))
+            return
+
+        logger.info("\t{}:".format(fname))
+
+        if self._is_json(metric):
```

Comment:
what was wrong with using `typ`, btw?

Reply:
I thought that I was breaking some boundaries by taking out type from the `show` method.

What I'm planning to do, is to return the output already formatted from `repo.metrics.show`, that way, I keep respecting the interface and have access to `typ`.

Kudos to @efiop for the suggestion ^